cloudfoundry/cfapi/service_manager.go

```diff
@@ -505,6 +507,97 @@ func (sm *ServiceManager) ReadServiceInstance(serviceInstanceID string) (service
 	return serviceInstance, nil
 }
 
+// WaitServiceInstanceTo -
+func (sm *ServiceManager) WaitServiceInstanceTo(operationType string, serviceInstanceID string) (err error) {
+	sm.log.UI.Say("Waiting for service instance %s to finish starting ..", terminal.EntityNameColor(serviceInstanceID))
+
+	c := make(chan error)
+	go func() {
+
+		var err error
+		var serviceInstance CCServiceInstance
+
+		for {
+			time.Sleep(appStatePingSleep)
+			if serviceInstance, err = sm.ReadServiceInstance(serviceInstanceID); err != nil {
+				c <- err
+				return
+			}
+
+			if serviceInstance.LastOperation["type"] == operationType {
+				state := serviceInstance.LastOperation["state"]
+
+				switch state {
+				case "succeeded":
+					c <- nil
+					return
+				case "failed":
+					c <- fmt.Errorf("service instance %s crashed", serviceInstanceID)
+					return
+				}
+			}
+		}
+	}()
+
+	select {
+	case err = <-c:
+		if err != nil {
+			return err
+		}
+		sm.log.UI.Say("%s finished starting ...", terminal.EntityNameColor(serviceInstanceID))
+	}
+	return nil
+}
+
+// WaitDeletionServiceInstance -
+func (sm *ServiceManager) WaitDeletionServiceInstance(serviceInstanceID string) (err error) {
+	sm.log.UI.Say("Waiting for service instance %s to delete ..", terminal.EntityNameColor(serviceInstanceID))
+	c := make(chan error)
+	go func() {
+
+		var err error
+		var serviceInstance CCServiceInstance
+
+		for {
+			time.Sleep(appStatePingSleep)
```

Comment:
Same, please  consider leveraging the [StateChangeConf](https://github.com/terraform-providers/terraform-provider-aws/blob/9fc15be7dd071e12e1e1468e998a070ed197d385/aws/resource_aws_db_instance.go#L527-L545) helper  using the same polling loop pattern than what the aws_db_instance does